gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSampling.java

```diff
@@ -134,16 +129,7 @@ public LogicalGraph sample(LogicalGraph graph) {
       .filter(new PageRankResultVertexFilter(
         threshold, sampleGreaterThanThreshold, keepVerticesIfSameScore));
 
-    DataSet<Edge> newEdges = graph.getEdges()
-      .join(scaledVertices)
-      .where(new SourceId<>()).equalTo(new Id<>())
-      .with(new LeftSide<>())
-      .join(scaledVertices)
-      .where(new TargetId<>()).equalTo(new Id<>())
-      .with(new LeftSide<>());
-
-    graph = graph.getFactory().fromDataSets(scaledVertices, newEdges);
-
-    return graph;
+    return graph.getFactory()
+      .fromDataSets(graph.getGraphHead(), scaledVertices, graph.getEdges()).verify();
```

Comment:
We decide in #1236 if we create a new graph head or not. Please remove the forwarding of the graph head. Also put the verify() call to the line below for better readability

Reply:
I would need to change the test significantly to remove all uses of graph head properties and the check, whether all vertices had the same score wouldn't be as reliable.

Reply:
So the test was invalid before? Or how the test works without passing the head?

Reply:
The head was passed before the sampling cleanup. After that the missing head caused the NullPointerException.

Reply:
omg ^^ Now we have to decide if we stay persistent and create a new graph head as in all other sampling operators (which needs a refactoring of the according test) OR we pass the graph head and violate the rule. @matthiastaeschner @galpha @rostam ... any opinions? Should the graph head of the sampled graph still having the necessary aggregated properties stored or should they are removed after successful sampling? I tend to the second.

Reply:
After a talk with @matthiastaeschner we decide to discard all properties of the graph head, because there are not needed after sampling. I know this makes testing a bit harder, but we stay persistent. If we decide later to use the existing graph head for sampling too, we have to remove all used properties from it.